### PR TITLE
Fix harmless bugs and Rewrite comments

### PR DESCRIPTION
**Pull request by wickdynex**
## Description

Fix some harmless bugs and rewrite some comments on the method.

For example:
- Use `equals` method to substitute `==` to compare `Integer` object.
- Rewrite method's comment to make it easier to use.
- Remove unnecessary logic to reduce code size and enhance execution time.

## Related Issue

None

## Type of Change

Please select the type of change you are making:

- [X] Bug fix (non-breaking change that fixes an issue)
- [X] Update logic 
- [X] Rewrite document

## Additional Information

I've run command `mvn test` to test this modification🥳, but can't pass all because the `postgresql` module was wrong🥺. As a result, you can merge it easily into `main` branch. If confused, plz contact to me. Looking forward to your reply🥰.

### CODE REVIEW

**Comment by dsyer** [comment]
LGTM. COuld you swqush it down to a single commit though please?

**Comment by dsyer** [comment]
You need to squash and rebase and force push to get it down to 1 commit, please. Github currently shows 5 commits.

File: src/main/java/org/springframework/samples/petclinic/owner/PetController.java [hunk 1]
```diff
@@ -138,7 +138,7 @@ public String processUpdateForm(@Valid Pet pet, BindingResult result, Owner owne
 		// checking if the pet name already exist for the owner
 		if (StringUtils.hasText(petName)) {
 			Pet existingPet = owner.getPet(petName.toLowerCase(), false);
-			if (existingPet != null && existingPet.getId() != pet.getId()) {
+			if (existingPet != null && existingPet.getId().equals(pet.getId())) {
```
Comment (dsyer) [inline src/main/java/org/springframework/samples/petclinic/owner/PetController.java:141]: This is the cause of the test failure (the logic is inverted).

**Comment by wickdynex** [comment]
> You need to squash and rebase and force push to get it down to 1 commit, please. Github currently shows 5 commits.

Oh, I've never used this `squash` feature before, and takes a little time to learn how to use🥺. And finally `squash` the 6 commits successfully, thanks for your reminder🥳. Btw, I fixed the `reversal of equal logic`. In my memory, I've fixed this bug, but perhaps due to the wrong operate to use `rebase` command, maybe I lost the correct modification😵‍💫.

Now, it words as expected like this:
image

Is there anything I can do for this PR? Looking forward to your reply, and also appreciate for your guide🥰.

**Pull request merged**